azure/container-linux/kubernetes/outputs.tf

```diff
@@ -54,3 +61,7 @@ output "backend_address_pool_id" {
   value       = azurerm_lb_backend_address_pool.worker.id
 }
 
+output "backend_probe_id" {
```

Comment:
For load balancing customization, you should add your own probe as shown [here](https://typhoon.psdn.io/architecture/azure/#tcpudp-services). Its more flexible and can accomodate if you're doing more advanced stuff like TCP/UDP (which likely should check different ports, no 10254). Even if you do aim to check Ingress, its better to define what you want directly, since I'm considering the builtin probe an internal detail that could change.

In your example, if you're using ingress to proxy node ports (or whatever "extra ports" are) you probably want the rule probing those extra ports. For example, what if you add another Nginx Ingress class that satisfies 10254, but not your extra ports. tldr, custom case => custom probe.

Reply:
that makes sense.